src/components/VmActions/WindowsRdpButton.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import { getRDP } from '../../actions'
+import { isWindows, hrefWithoutHistory } from '../../helpers'
+import { canConsole } from '../../vm-status'
+
+/**
+ * Button to send a RDP connection file to the user for a Windows VM.
+ */
+const WindowsRdpButton = ({ vm, className, config, onRDP }) => {
+  const isWindowsVM = isWindows(vm.getIn(['os', 'type']))
+
+  let component = null
+  if (isWindowsVM && canConsole(vm.get('status'))) {
+    const idPrefix = `consoleaction-${vm.get('name')}`
+    const vmName = vm.get('name')
+    const fqdn = vm.get('fqdn')
+    const domain = config.get('domain')
+    const username = config.getIn([ 'user', 'name' ])
+
+    component = (
+      <a
+        href='#'
+        key={vm.get('id')} id={`${idPrefix}-rdp`}
+        className={className}
+        onClick={hrefWithoutHistory(onRDP(vmName, fqdn, domain, username))}
+      >
+        RDP
+      </a>
+    )
+  }
+
+  return component
+}
+WindowsRdpButton.propTypes = {
+  vm: PropTypes.object.isRequired,
+  className: PropTypes.string.isRequired,
+
+  config: PropTypes.object.isRequired,
+  onRDP: PropTypes.func.isRequired,
+}
+
+export default connect(
+  (state) => ({
+    config: state.config,
+  }),
+  (dispatch) => ({
+    onRDP: (vmName, fqdn, domain, username) => dispatch(getRDP({ vmName, fqdn, domain, username })),
```

Comment:
I think we can get `vmName` and `fqdn` from `vm` as `ownProps`.

Reply:
Done!